src/iOS.Core/Utilities/ASHelpers.cs

```diff
@@ -135,21 +136,40 @@ public static async Task<bool> IsAutofillEnabledAsync()
 
         public static IASCredentialIdentity? ToCredentialIdentity(CipherView cipher)
         {
-            if (!cipher.HasFido2Credential)
+            try
             {
-                return ToPasswordCredentialIdentity(cipher);
-            }
+                if (!cipher.HasFido2Credential)
+                {
+                    return ToPasswordCredentialIdentity(cipher);
+                }
+
+                if (!cipher.Login.MainFido2Credential.DiscoverableValue)
+                {
+                    return null;
+                }
+
+                var userName = cipher.Login.MainFido2Credential.UserName;
+                if (string.IsNullOrWhiteSpace(userName))
+                {
+                    userName = cipher.Login.MainFido2Credential.UserDisplayName;
 
-            if (!cipher.Login.MainFido2Credential.DiscoverableValue)
+                    if (string.IsNullOrWhiteSpace(userName))
+                    {
+                        userName = cipher.Login.Username;
+                    }
+                }
```

Comment:
suggestion: I think you can skip the nesting

```suggestion
                if (string.IsNullOrWhiteSpace(userName))
                {
                    userName = cipher.Login.MainFido2Credential.UserDisplayName;
                }
                
                if (string.IsNullOrWhiteSpace(userName))
                {
                    userName = cipher.Login.Username;
                }
```

Reply:
ℹ️ I did it on purpose to avoid checking twice for the same thing on the primary flow, i.e. if the most probable thing happens that is `UserName` not being null then first time it checks we know it's not null then it's not needed to check again afterwards the same thing. Nonetheless it's necessary in the nesting for the alternative case.
If we want to avoid nesting I could extract this into a method where we can just return the value on each case without further nesting, but it's small and just one additional level so I don't thing is so necessary.

Reply:
Oh interesting, I didn't consider that, fair point!